crates/relayer/src/config.rs

```diff
@@ -556,14 +558,37 @@ pub enum EventSourceMode {
 
 #[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
 #[serde(deny_unknown_fields)]
-pub struct ChainConfig {
+#[serde(tag = "type")]
+pub enum ChainConfig {
+    CosmosSdk(CosmosSdkConfig),
+}
+
+impl ChainConfig {
+    pub fn id(&self) -> &ChainId {
+        match self {
+            Self::CosmosSdk(config) => &config.id,
+        }
+    }
+
+    pub fn packet_filter(&self) -> &PacketFilter {
+        match self {
+            Self::CosmosSdk(config) => &config.packet_filter,
+        }
+    }
+
+    pub fn max_block_time(&self) -> Duration {
+        match self {
+            Self::CosmosSdk(config) => config.max_block_time,
+        }
+    }
+}
```

Comment:
These are the attributes that are used by Hermes outside of the Cosmos backend.

All of the other attributes are only used inside the Cosmos backend, so this is effectively a way to discover which parts of the configuration data is used by the relayer engine, and which parts are used only by the chain backend.

All other backends' configuration objects will need these fields.

An alternative representation would be to have the `ChainConfig` contain these common fields unconditionally, AND an inner enum with only the non-common configuration data. But I think this is actually worse, because now the backends can't cleanly model their own data: they need access to common fields like the chain id, and those fields would live in an overarching `ChainConfig` object. With the current structure, the common fields would be duplicated across different objects (the `PenumbraConfig` would have an `id` field as well), but this actually makes the code more modular, because each backend can just pass around and work with one object it defines (the cosmos logic can work with a `CosmosSdkConfig`, the penumbra logic can work with a `PenumbraConfig`, and so on).